contracts/libraries/Offer.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+string constant OFFER_TYPESTRING = "Offer(address taker,address maker,address takerToken,uint256 takerTokenAmount,address makerToken,uint256 makerTokenAmount,uint256 minMakerTokenAmount,uint256 expiry,uint256 salt)";
+
+bytes32 constant OFFER_DATA_TYPEHASH = 0x08a878f9010ee633419804c459e9e6107b7679cf23b9efe0605475501bdb5d18;
+// keccak256(OFFER_TYPESTRING);
+
+struct Offer {
+    address taker;
+    address payable maker;
+    address takerToken;
+    uint256 takerTokenAmount;
+    address makerToken;
+    uint256 makerTokenAmount;
+    uint256 minMakerTokenAmount;
+    uint256 expiry;
+    uint256 salt;
+}
+
+// solhint-disable-next-line func-visibility
+function getOfferHash(Offer memory offer) pure returns (bytes32) {
```

Comment:
Q: What do you think about using the library for struct pattern, to organize the hash relevant work and avoid the hash scattered many places? Like

```solidity
// libraries/Offer.sol
struct Offer { ... }

library OfferLib {
    public constant OFFER_TYPESTRING
    public constant OFFER_DATA_TYPEHASH

    function getHash(...) { ... }
}

// Contract
contract A {
    using OfferLib for Offer
}
```

I think this can also apply to strategy specific struct, but may need additional effort to put the struct in the right place. A rough and possible ways:

* In one file
```solidity
// interfaces/IRFQ.sol
interface IRFQ { ... }

struct RFQOrder { ... }

library RFQOrderLib { ... }
```

* In separate file
```solidity
// interfaces/IRFQ.sol
import { RFQOrder } from "libraries/RFQOrder.sol"

interface IRFQ { ... }

// libraries/RFQOrder.sol
struct RFQOrder { ... }

library RFQOrderLib { ... }
```

Would like to know more opinions and you guys' preference. 😃

Reply:
I'm ok with making other struct/type into a single file like `Offer.sol`. We can go back and adjust it in the future if needed.

Reply:
How about put this refactoring in another PR?